client/web/src/enterprise/executors/secrets/backend.ts

```diff
@@ -0,0 +1,287 @@
+    ExecutorSecretAccessLogFields,
+    ExecutorSecretAccessLogsResult,
+    ExecutorSecretAccessLogsVariables,
+} from '../../../graphql-operations'
+
+const EXECUTOR_SECRET_FIELDS = gql`
+    fragment ExecutorSecretFields on ExecutorSecret {
+        id
+        key
+        scope
+        overwritesGlobalSecret
+        createdAt
+        updatedAt
+        creator {
+            id
+            username
+            displayName
+            url
+        }
+        namespace {
+            id
+            namespaceName
+            url
+        }
+    }
+`
+
+export const CREATE_EXECUTOR_SECRET = gql`
+    mutation CreateExecutorSecret($scope: ExecutorSecretScope!, $namespace: ID, $key: String!, $value: String!) {
+        createExecutorSecret(scope: $scope, namespace: $namespace, key: $key, value: $value) {
+            ...ExecutorSecretFields
+        }
+    }
+
+    ${EXECUTOR_SECRET_FIELDS}
+`
+
+export const useCreateExecutorSecret = (): MutationTuple<CreateExecutorSecretResult, CreateExecutorSecretVariables> =>
+    useMutation(CREATE_EXECUTOR_SECRET)
+
+export const UPDATE_EXECUTOR_SECRET = gql`
+    mutation UpdateExecutorSecret($scope: ExecutorSecretScope!, $secret: ID!, $value: String!) {
+        updateExecutorSecret(scope: $scope, id: $secret, value: $value) {
+            ...ExecutorSecretFields
+        }
+    }
+
+    ${EXECUTOR_SECRET_FIELDS}
+`
+
+export const useUpdateExecutorSecret = (): MutationTuple<UpdateExecutorSecretResult, UpdateExecutorSecretVariables> =>
+    useMutation(UPDATE_EXECUTOR_SECRET)
+
+export const DELETE_EXECUTOR_SECRET = gql`
+    mutation DeleteExecutorSecret($scope: ExecutorSecretScope!, $id: ID!) {
+        deleteExecutorSecret(scope: $scope, id: $id) {
+            alwaysNil
+        }
+    }
+`
+
+export const useDeleteExecutorSecret = (): MutationTuple<DeleteExecutorSecretResult, DeleteExecutorSecretVariables> =>
+    useMutation(DELETE_EXECUTOR_SECRET)
+
+const EXECUTOR_SECRET_CONNECTION_FIELDS = gql`
+    fragment ExecutorSecretConnectionFields on ExecutorSecretConnection {
+        totalCount
+        pageInfo {
+            hasNextPage
+            endCursor
+        }
+        nodes {
+            ...ExecutorSecretFields
+        }
+    }
+
+    ${EXECUTOR_SECRET_FIELDS}
+`
+
+export const USER_EXECUTOR_SECRETS = gql`
+    query UserExecutorSecrets($user: ID!, $scope: ExecutorSecretScope!, $first: Int, $after: String) {
+        node(id: $user) {
+            __typename
+            ... on User {
+                executorSecrets(scope: $scope, first: $first, after: $after) {
+                    ...ExecutorSecretConnectionFields
+                }
+            }
+        }
+    }
+
+    ${EXECUTOR_SECRET_CONNECTION_FIELDS}
+`
+
+export const userExecutorSecretsConnectionFactory = (
+    user: Scalars['ID'],
+    scope: ExecutorSecretScope
+): UseConnectionResult<ExecutorSecretFields> =>
+    // Scope has to be injected dynamically.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
I feel like I'm missing something. Why does this seem to need a different treatment from `useExecutorSecretAccessLogsConnection` down below? That one injects the variable `secret` the exact same way, no?  I would have expected this to just be defined like:

```suggestion
export const useUserExecutorSecretsConnection = (
    user: Scalars['ID'],
    scope: ExecutorSecretScope
): UseConnectionResult<ExecutorSecretFields> =>
```

I think the linter warning is just over a hook being called from a non-component, non-hook-function, right? I think just naming it in the "hook-y" way (starts with `use`) is all that's going on here.

Reply:
Yes but then it complains about the use in `useCallback` above, which is required because the child list component injects the scope to the parent using that. 
It's a bit weirdly coupled sadly.